lighthouse-core/gather/fetcher.js

```diff
@@ -84,19 +90,57 @@ class Fetcher {
   }
 
   /**
-   * Requires that `driver.enableRequestInterception` has been called.
+   * Requires that `fetcher.enable` has been called.
    *
    * Fetches any resource in a way that circumvents CORS.
    *
    * @param {string} url
-   * @param {{timeout: number}} options timeout is in ms
+   * @param {{timeout: number}=} options timeout is in ms
    * @return {Promise<string>}
    */
-  async fetchResource(url, {timeout = 500}) {
+  async fetchResource(url, options = {timeout: 500}) {
     if (!this._enabled) {
-      throw new Error('Must call `enableRequestInterception` before using fetchResource');
+      throw new Error('Must call `enable` before using fetchResource');
+    }
+
+    const milestone = await this.driver.getBrowserVersion().then(v => v.milestone);
+    if (milestone >= 88) {
+      return await this._fetchResourceOverProtocol(url, options);
+    }
+    return await this._fetchResourceIframe(url, options);
+  }
+
+  /**
+   * @param {string} url
+   * @param {{timeout: number}=} options timeout is in ms
+   * @return {Promise<string>}
+   */
+  async _fetchResourceOverProtocol(url, options = {timeout: 500}) {
```

Comment:
this timeout doesn't apply to the actual network request, though, which seems like more of a worry than streaming the response from the backend?

Reply:
We could set the timeout on fetching the resource, then set the remaining time as the timeout for reading the IO stream.

Reply:
@brendankenny What do you think of the new timeout structure?